packages/near-membrane-base/src/connector.ts

```diff
@@ -6,10 +6,111 @@ const marshallSourceTextInStrictMode = `
+                        if (
+                            typeof funcName === 'string' &&
+                            funcName !== 'eval' &&
+                            ReflectApply(StringProtoEndsWith, funcName, [LOCKER_IDENTIFIER_MARKER])
+                        ) {
+                            isMarked = true;
+                        }
+                        if (!isMarked) {
+                            const evalOrigin = ReflectApply(
+                                CallSiteProtoGetEvalOrigin,
+                                callSite,
+                                []
+                            );
+                            if (
+                                typeof evalOrigin === 'string' &&
+                                ReflectApply(StringProtoIncludes, evalOrigin, [
+                                    LOCKER_IDENTIFIER_MARKER,
+                                ])
+                            ) {
+                                isMarked = true;
+                            }
+                        }
+                        // Only write a single LWS entry per consecutive LWS stacks.
+                        if (isMarked) {
+                            if (!consecutive) {
+                                consecutive = true;
+                                stackTrace += '\n    at LWS';
+                            }
+                            continue;
+                        } else {
+                            consecutive = false;
+                        }
+                        try {
+                            stackTrace += `\n    at ${ReflectApply(
+                                CallSiteProtoToString,
+                                callSite,
+                                []
+                            )}`;
+                            // eslint-disable-next-line no-empty
+                        } catch {}
+                    }
+                    return stackTrace;
+                };
+                // Make Error.prepareStackTrace non-configurable and non-writable.
+                Reflect.defineProperty(Error, 'prepareStackTrace', {
+                    // @ts-ignore: TS doesn't like __proto__ on property descriptors.
+                    __proto__: null,
+                    enumerable: true,
+                    // Error.prepareStackTrace cannot be a bound or proxy wrapped
+                    // function, so to obscure its source we wrap the call to
+                    // formatStackTrace().
+                    value: function prepareStackTrace(error: Error, callSites: NodeJS.CallSite[]) {
+                        return formatStackTrace(error, callSites);
+                    },
+                });
+            }
+            if (typeof Error.stackTraceLimit === 'number') {
+                // The default stack trace limit is 10.
+                // Increasing to 20 for wiggle room of filtered results.
+                Error.stackTraceLimit *= 2;
```

Comment:
Should this be configurable?

Reply:
I think it's fine if users want to tweak how long the call stack is... I can make it configurable in debug mode though.